handsontable/types/plugins/emptyDataState/emptyDataState.d.ts

```diff
@@ -0,0 +1,24 @@
+import Core from '../../core';
+import { BasePlugin } from '../base';
+
+interface EmptyDataStateMessage {
+  title?: string;
+  description?: string;
+  actions?: {
+    text: string;
+    type: 'primary' | 'secondary';
+    callback: () => void;
+  }[];
+}
+
+export interface EmptyDataStateConfig {
+  message?: string | ((source: string) => EmptyDataStateMessage | undefined) | EmptyDataStateMessage;
```

Comment:
The `source` could be defined as a type `empty` and `filters` eventually `unknown` for unknown callers? 🤔